packages/shared/lib/utils/utils.ts

```diff
@@ -79,8 +79,15 @@ export function parseTokenExpirationDate(expirationDate: any): Date | undefined
         return expirationDate;
     }
 
-    // UNIX timestamp
     if (typeof expirationDate === 'number') {
+        const dateFromValue = new Date(expirationDate);
+        // Check if expirationDate is in milliseconds
+        const isMs = dateFromValue.getFullYear() > 1971;
+        const currentTime = Date.now();
+        if (isMs) {
+            return new Date(currentTime + expirationDate);
+        }
```

Comment:
Are you sure about this logic?

When we receive expirationDate, it's a fully fledged timestamp, no? So why would you add currentTime on top of it?